### ISSUE TITLE
<instrumentation> and subelements should allow att.edit

### ISSUE THREAD

**Issue by ahankinson** [music-encoding/music-encoding#227]
_From [klaus.re...@gmail.com](https://code.google.com/u/105834722588578911678/) on January 29, 2015 08:11:10_

At some point it may not be possible to identify the original instrumation of a work (completely). So it should be possible to mark the coding with @-cert.

_Original issue: http://code.google.com/p/music-encoding/issues/detail?id=227_

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on January 29, 2015 07:52:08_

Examples of the use of att.edit on <instrumentation> and its sub-elements would be useful.

**Comment by ahankinson** [comment]
_From [klaus.re...@gmail.com](https://code.google.com/u/105834722588578911678/) on February 02, 2015 09:13:59_

If there are no musical sources for a piece, which is only known to be a cantata maybe it could be coded like: 

<instrumentation cert="unknown">
  <instrVoiceGrp>
    <head>Choir</head>
  </instrVoiceGrp>
  <instrVoiceGrp>
    <head>Orchestra</head>
  </instrVoiceGrp>
</instrumentation>

Or, like in my case, there is one manuscript score. But a catalogue entry (not for the known manuscript) states a larger orchestra. I would propose:

<instrumentation>
  <instrVoiceGrp xml:id="InstrGrp">
    <head>Orchestra</head>
    <instrVoice count="2" cert="medium" xml:id="cor">Corno</instrVoice>
    <instrVoice count="2">Oboi</instrVoice>
    <instrVoice count="2">Violino</instrVoice>
    <instrVoice count="1" cert="medium" xml:id="va">Viola</instrVoice>
    <instrVoice count="1">Basso continuo</instrVoice>
  </instrVoiceGrp>
  <instrVoiceGrp xml:id="VoiceGrp">
    <head>Choir SATB</head>
    <instrVoice count="1">S</instrVoice>
    <instrVoice count="1">A</instrVoice>
    <instrVoice count="1">T</instrVoice>
    <instrVoice count="1">B</instrVoice>
  </instrVoiceGrp>
</instrumentation>

Details could be given somewhere else, for example within the appropriate <source> using the xml:ids.

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on February 02, 2015 13:07:35_

If what you're trying to accomplish is to differentiate between the instrumentation given by the manuscript vs. what is given by a catalog entry (presumably for the work), then the instrumentation for the work can be recorded in work/perfMedium/instrumentation (or in expression/perfMedium/instrumentation, if the FRBR module is activated) and a note created in either source or item stating that the particular manifestation/source or this particular item is lacking the horn and viola parts/staves.

A more algorithmically processable option is to allow perfMedium/instrumentation in source and item. This subverts the usage of the FRBR module, though.  I'm not entirely against that when it's appropriate, just calling attention to it. :-)

Axel probably has some thoughts on this issue.

**Comment by ahankinson** [comment]
_From [a...@kb.dk](https://code.google.com/u/101822190470135414035/) on February 03, 2015 03:40:06_

If I understand this right there are two scenarios discussed here: One where the intended instrumentation of the work is uncertain because of lack of information and one where different instrumentations are used in the sources (of course, a missing part in a set of parts is not a different instrumentation; it is just an incomplete source - that's another thing).
In general, I try to avoid using @cert because I don't find the possible values - low, high, medium, unknown - very informative. Except for "unknown" they signify a judgement but give no information on the evidence they are based on. Combining it with @evidence helps in some cases. For the instrumentation of a lost or incompletely preserved work, @-cert="unknown" may be useful, though, such as in Klaus' first example. 
The other example given is a different: It lists an instrumentation as assumed by the editor, if I understand you correctly. In this case I would find it more useful to indicate that the assumed instrumentation was added by the editor. 
Why not allow <add> in <perfMedium> and its descendants? In that case, you could use:

<instrumentation>
  <instrVoiceGrp xml:id="InstrGrp">
    <head>Orchestra</head>
    <add evidence="conjecture"><instrVoice count="2">Corno</instrVoice></add>
    <instrVoice count="2">Oboi</instrVoice>
    <instrVoice count="2">Violino</instrVoice>
    <add evidence="conjecture"><instrVoice count="1">Viola</instrVoice></add>
    <instrVoice count="1">Basso continuo</instrVoice>
  </instrVoiceGrp>
  [...]
</instrumentation>

I am not sure whether the use of perfMedium in <source> or <item> would actually subvert the FRBR module. Conflicting instrumentations across FRBR levels are unavoidable: I believe that it does make sense to say that the instrumentation 
of a particular symphony is a large orchestra even when some of its sources present a different one - for instance, a piano score. Otherwise it would be impossible to indicate a work's instrumentation at all, but only instrumentations at source or item level. Or: every variation in instrumentation would have to constitue a new work. That's probably not what everyone wants either.

A third situation is where a score contains staves without indication of instrument. With <add> allowed within <instrVoice>, you could do this:

<instrVoice count="2">
  <add evidence="conjecture">Oboi</add>
</instrVoice>

**Comment by ahankinson** [comment]
_From [klaus.re...@gmail.com](https://code.google.com/u/105834722588578911678/) on February 03, 2015 05:45:27_

Indeed I was thinking of the work as abstract idea rather than the various sources. So the last example of Axel would be more source specific. (And I quite like the <add> in there.) But, to clarify this a little, for the instrumentation of a !work! you have to consider every piece of information, some from actual musical sources, others from mentioning in letters or a lexicon. Basically you would have instruments named in every source and others just once said to be part of it. To bring this together without downgrading or discrediting a specific source (as editorial decision) I guess it would be best having a @cert and @-xml:id with verbose statements at some other point.

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on February 23, 2015 12:41:18_

Pursuing Axel's differentiation of 2 use cases of instrumentation -- at the work level and at the expression level (MEI _doesn't_ currently allow instrumentation at the manifestation level) -- it seems to me that a _work_ can ultimately have only 1 statement of its instrumentation.  This statement must be a synthesis of all the information at "lower"; that is, expression, manifestation (source) and item, levels. Any attempt to show via markup how that synthesis was accomplished seems to me to be doomed to failure due to its complexity.

I am willing, however, to concede the need for an <annot> inside <instrumentation> to permit recording of the process used to arrive at the statement of instrumentation and other pertinent info. In addition, I think it would be beneficial to make <instrumentation> and its descendants (ensemble, instrVoice, and instrVoiceGrp) members of att.edit. This brings @cert, @evidence, and @-resp to bear on the problem.

The use of these attributes on work/instrumentation/ and its children would allow capture of a certain level of ambiguity without the introduction of elements like <add> that, while useful for capturing source text, are difficult to make sense of in a data-centric way.

For example, 

<instrumentation evidence="conjecture" resp="#kr">
  <instrVoiceGrp>
    <head>Orchestra</head>
    <instrVoice count="2">Oboi</instrVoice>
    <instrVoice count="2">Violino</instrVoice>
    <instrVoice count="1">Basso continuo</instrVoice>
  </instrVoiceGrp>
  [...]
</instrumentation>

says the entire instrumentation statement is conjecture on Klaus' part, while 

<instrumentation>
  <instrVoiceGrp>
    <head>Orchestra</head>
    <instrVoice count="2" evidence="conjecture" resp="#kr">Corno</instrVoice>
    <instrVoice count="2">Oboi</instrVoice>
    <instrVoice count="2">Violino</instrVoice>
    <instrVoice count="1" evidence="conjecture" resp="#kr">Viola</instrVoice>
    <instrVoice count="1">Basso continuo</instrVoice>
  </instrVoiceGrp>
  [...]
</instrumentation>

says he thinks there also ought to be horns and violas.  Using attributes like this along with a textual explanation allowed by <annot> should be sufficient.

This will also suffice at the expression (or work) level to indicate variation in the instrumentation across sources --

<instrumentation>
  <instrVoiceGrp>
    <head>Orchestra</head>
    <instrVoice count="2" source="source2">Corno</instrVoice>
    <instrVoice count="2" source="source1 source2">Oboi</instrVoice>
    <instrVoice count="2" source="source1 source2">Violino</instrVoice>
    <instrVoice count="1" source="source2">Viola</instrVoice>
    <instrVoice count="1" source="source1 source2">Basso continuo</instrVoice>
  </instrVoiceGrp>
  [...]
</instrumentation>

which states that the oboes, violins and basso continuo are accounted for in both sources, but the horns and violas are only present in source2.

While we're at it, we could also allow <instrumentation> within <source>.  BUT, this is the thing that I feel would "subvert" FRBR.  For instance, if one can specify instrumentation for each source, what is the point of using <expression>?  I'm not averse to the idea, however.  I can definitely see the usefulness is allowing <instrumentation> within <source>.  It's just that this offers no incentive for one to use the FRBR capabilities of MEI.  Currently, because <instrumentation> is _not available_ in <source>, one is guided toward (forced?) to create multiple <expression> elements, to which one or more sources can be linked.

**Comment by ahankinson** [comment]
_From [a...@kb.dk](https://code.google.com/u/101822190470135414035/) on February 24, 2015 07:10:05_

Making <instrumentation> and descendants members of att.edit is fine with me as an alternative to <add> since @-evidence="conjecture" is enough to indicate that the content has been added.

The question about <instrumentation> in <source> is a complex one, which I think deserves discussion on MEI-L rather than here. At the MEC2013 I tried to make the argument that the very narrow use of the expression level apparently suggested in the IFLA's FRBR report would result more or less in a 1-to-1 relationship between expression and manifestation, because - but only in principle - even the smallest revision or difference would result in a new expression. In this case the question would rather be: what is the point of using <source>? This is especially true when dealing with manuscripts, where usually there is also a 1-to-1 relationship between manifestation (=source) and item. This is why I would suggest interpreting expressions as versions in a more comprehensive way (remembering that expressions are abstract entities). Putting varying instrumentations as in the examples above into two different expressions may be fine in that particular case: We have two different versions of the work (provided that we are talking about scores, not two originally complete sets of parts with different parts missing). But what about a short score, such as a sketch for a symphony, for instance? Or a piano score or short intended for rehearsal of the orchestral version? I would be reluctant to define each of these as distinct expressions, because I would lose the possibility to group them together as one version as opposed to, for instance, a later revision of the work. 
This rather lengthy reply may seem slightly off the point because – as Perry demonstrates – it is quite possible to indicate the instrumentation of the individual sources at expression level. My point is, however, that <instrumentation> at source level may be a more explicit and in my opinion more elegant way of encoding this variation, keeping information about the individual sources within <source> instead of extracting it to expression level.
